tamr_toolbox/data_io/core_connect.py

```diff
@@ -0,0 +1,85 @@
+"""Tasks related to interacting with the Tamr Core-connect"""
+import json
+import logging
+from tamr_toolbox.models.data_type import JsonDict
+from tamr_unify_client import Client
+from tamr_toolbox.utils import version
+
+LOGGER = logging.getLogger(__name__)
+
+
+def jdbc_ingest(
+    *,
+    client: Client,
+    jdbc_connect: JsonDict,
+    dataset_name: str,
+    query: str,
+    primary_key=None,
+    truncate_tamr_dataset=None,
+    retrieve_connect_metadata=None,
+    retrieve_source_metadata=None,
+    tamr_min_version="2022.005.0",
+) -> JsonDict:
+    """
+    Ingest a dataset into Tamr via core_connect given query config, dataset name, query string,
+    and optional list of columns for primary key
+
+    Args:
+        client: Tamr unify client object.
+        jdbc_connect: queryConfig used by the API call.
+        dataset_name: Name of Tamr dataset.
+        query: jdbc query to execute in the database and results of which will be loaded into Tamr.
+        primary_key(optional): list of columns to use as primary key.
+        truncate_tamr_dataset(optional): Truncate the Tamr dataset if it exists.
+        retrieve_connect_metadata(optional): Core Connect retrieves the metadata for services.
+        retrieve_source_metadata(optional): Only applies to Snowflake.
+        tamr_min_version(optional): The minimum version of Tamr to use Core Connect.
+
+    Raises:
+        HTTPError: if the call to ingest the dataset was unsuccessful
+    """
+
+    # Check Tamr version
+    if version.is_version_condition_met(
+        tamr_version=version.current(client), min_version=tamr_min_version
+    ):
+        None
+    else:
+        error_message = "Core-connect is not available in current version of Tamr."
+        LOGGER.error(error_message)
+        raise Exception(error_message)
+
+    # handle primary key
+    if primary_key is None:
+        primary_key = []
+    else:
+        primary_key = primary_key.split(",")
+
+    # ingest data
+    api_path = "/api/connect/jdbcIngest/ingest"
```

Comment:
Add the async=true argument to the URL and include job monitoring using the jobs APIs. Feel free to copy my existing function: https://github.com/Datatamer/ThermoFisher/blob/master/airflow_plugins/unify_plugin/playbooks/thermo_fisher_SpendHQ_import.py#L67